python/objToJSON.c

```diff
@@ -887,6 +896,69 @@ PyObject* objToJSON(PyObject* self, PyObject *args, PyObject *kwargs)
     encoder.rejectBytes = orejectBytes;
   }
 
+  if (oseparators != NULL && oseparators != Py_None)
+  {
+    if (!PyTuple_Check(oseparators))
+    {
+      PyErr_SetString(PyExc_TypeError, "expected tuple or None as separator");
+      return NULL;
+    }
+    if (PyTuple_Size (oseparators) != 2)
+    {
+      PyErr_SetString(PyExc_ValueError, "expected tuple of size 2 as separator");
+      return NULL;
+    }
+    oseparatorsItem = PyTuple_GetItem(oseparators, 0);
+    if (PyErr_Occurred())
+    {
+      return NULL;
```

Comment:
Surely this line isn't reachable if the check above has already confirmed that there are two separators?

Reply:
You're right that it's currently impossible to hit this given CPython's implementation of `PyTuple_GetItem`. But there's no guarantee that it couldn't fail for reasons unrelated to the index not matching the tuple's size...
The check should probably be `if (!oseparatorsItem)` (or `if (oseparatorsItem == NULL)`, existing code is inconsistent) though, which should perform better as well, although that's unlikely to matter in the setup function.